common/v2/features/BroadcastTransaction/stateFactory.tsx

```diff
@@ -0,0 +1,148 @@
+import { ITxReceipt, ITxConfig, ISignedTx, NetworkId } from 'v2/types';
+import { DEFAULT_NETWORK } from 'v2/config';
+import { translateRaw } from 'v2/translations';
+import { NetworkContext, getAssetByUUID, AssetContext } from 'v2/services/Store';
+import { fromWei, ProviderHandler } from 'v2/services/EthService';
+import { ToastContext } from 'v2/features/Toasts';
+
+const broadcastTxInitialState: State = {
+  network: DEFAULT_NETWORK,
+  txReceipt: undefined,
+  txConfig: undefined,
+  signedTransaction: '',
+  transaction: undefined,
+  networkSelectError: ''
+};
+
+interface State {
+  network: NetworkId;
+  txConfig: ITxConfig | undefined;
+  txReceipt?: ITxReceipt;
+  signedTransaction: ISignedTx;
+  networkSelectError: string;
+  transaction: EthTx | undefined;
+}
+
+const BroadcastTxConfigFactory: TUseStateReducerFactory<State> = ({ state, setState }) => {
+  const { networks, getNetworkByChainId, getNetworkByName } = useContext(NetworkContext);
+  const { assets } = useContext(AssetContext);
+  const { displayToast, toastTemplates } = useContext(ToastContext);
+
+  const handleNetworkChanged = (network: NetworkId) => {
+    setState((prevState: State) => ({
+      ...prevState,
+      network
+    }));
+  };
+
+  const handleSendClicked = (cb: any) => {
+    const { transaction, network } = state;
+    if (!transaction) return;
+
+    const { to, value, gasPrice, gasLimit, nonce, data } = transaction;
+    const chainId = transaction.getChainId();
+    const from = transaction.getSenderAddress();
+
+    const txNetwork = chainId ? getNetworkByChainId(chainId) : getNetworkByName(network);
+
+    if (!txNetwork) {
+      setState((prevState: State) => ({
+        ...prevState,
+        networkSelectError: translateRaw('BROADCAST_TX_INVALID_CHAIN_ID')
+      }));
+      return;
+    }
+
+    const txAmount = fromWei(new BN(value, 16), 'ether');
+    const txBaseAsset = getAssetByUUID(assets)(txNetwork.baseAsset)!;
+    const txToAddress = toChecksumAddressByChainId(to.toString('hex'), chainId);
+    const txFromAddress = toChecksumAddressByChainId(from.toString('hex'), chainId);
+    const txGasPrice = new BN(gasPrice, 16).toString();
+    const txGasLimit = new BN(gasLimit, 16).toString();
+    const txValue = new BN(value, 16).toString();
+    const txNonce = new BN(nonce, 16).toString();
+    const txData = addHexPrefix(data.toString('hex'));
+
+    const txConfig: any = {
+      amount: txAmount,
+      receiverAddress: txToAddress,
+      senderAccount: { address: txFromAddress, assets: [] },
+      network: txNetwork,
+      asset: txBaseAsset,
+      baseAsset: txBaseAsset,
+      gasPrice: txGasPrice,
+      gasLimit: txGasLimit,
+      value: txValue,
+      nonce: txNonce,
+      data: txData
+    };
+
+    setState((prevState: State) => ({
+      ...prevState,
+      networkSelectError: '',
+      txConfig
+    }));
+
+    cb();
+  };
+
+  const handleSignedTxChanged = (signedTransaction: string) => {
```

Comment:
it might be better to have this as an on-submit trigger and just add validation to input field (isHexString && isCorrectNumberOfCharacters). This would enable you to do the above comment as well ^.

Reply:
That makes sense, moved changeHandler from stateFactory to component together with signature validation.